db/db_engine.py

```diff
@@ -0,0 +1,171 @@
+import os
+from datetime import datetime
+
+from sqlalchemy import Enum, ForeignKey, String, DateTime, create_engine
+from sqlalchemy.orm import Mapped, mapped_column, relationship, sessionmaker, scoped_session, declarative_base
+from typing import List, Optional
+from uuid import uuid4
+
+DB_PATH = "db"
+DB_NAME = "database.db"
+engine = create_engine(rf"sqlite:///{os.path.join(os.path.join(__file__, '..', '..', DB_PATH, DB_NAME))}", echo=True)
+
+# Create a scoped session to manage sessions for each request
+Session = scoped_session(sessionmaker(bind=engine))
+
+# Base class for the models
+Base = declarative_base()
+
+
+class UploadStatus:
+    done = "done"
+    pending = "pending"
+
+
+def generate_uid() -> str:
+    """
+    Generates a unique identifier (UID) using the UUID4 algorithm.
+    Returns:
+        str: The generated unique identifier.
+    """
+    return str(uuid4())
+
+
+class User(Base):
+    """
+    Represents a User entity in the database.
+    Attributes:
+        id (int): The primary key for the User table.
+        email (str): The email address of the user.
+        uploads (List[Upload]): A list of uploads associated with the user.
+    """
+    __tablename__ = "user"
+    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    email: Mapped[str] = mapped_column(unique=True, nullable=False)
+    uploads: Mapped[List["Upload"]] = relationship(back_populates="user", cascade="all, delete-orphan")
+
+
+class Upload(Base):
+    """
+    Represents an Upload entity in the database.
+    Attributes:
+        id (int): The primary key for the Upload table.
+        uid (str): The unique identifier for the upload.
+        filename (str): The original filename of the uploaded file.
+        upload_time (DateTime): The timestamp of when the Web API received the upload.
+        finish_time (Optional[DateTime]): The timestamp of when the Explainer finished processing the upload.
+        status (str): The current status of the upload (either 'pending' or 'done').
+        user_id (Optional[int]): The foreign key referencing the User table,
+        indicating the user who uploaded this upload.
+    """
+    __tablename__ = "Upload"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    uid: Mapped[str] = mapped_column(String(36), default=generate_uid, nullable=False, unique=True)
```

Comment:
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.